src/pru-hardware-interface.h

```diff
@@ -0,0 +1,60 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * (c) 2013, 2016 Henner Zeller <h.zeller@acm.org>
+ *
+ * This file is part of BeagleG. http://github.com/hzeller/beagleg
+ *
+ * BeagleG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BeagleG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BeagleG.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef BEAGLEG_PRU_HARDWARE_INTERFACE_
+#define BEAGLEG_PRU_HARDWARE_INTERFACE_
+
+#include <cstddef>
+#include <pruss_intc_mapping.h>
```

Comment:
In this abstraction, this PRU specific include should probably not be here.

Think of it this way: everything should be compilable with some other implementation without even the am335x package in the source path.